infra/modules/azure_function_app_exposed/function_app_slot.tf

```diff
@@ -0,0 +1,63 @@
+resource "azurerm_linux_function_app_slot" "this" {
+  count = local.function_app.is_slot_enabled
+
+  name            = local.function_app_slot.name
+  function_app_id = azurerm_linux_function_app.this.id
+
+  storage_account_name          = azurerm_storage_account.this.name
+  storage_uses_managed_identity = true
+  builtin_logging_enabled       = false
+
+  https_only                    = true
+  public_network_access_enabled = true
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  site_config {
+    http2_enabled                          = true
+    always_on                              = true
+    vnet_route_all_enabled                 = true
+    application_insights_connection_string = var.application_insights_connection_string
```

Comment:
shouldn't we check `local.application_insights.enable` ?

Reply:
It is actually the same
```hcl
  application_insights = {
    enable = var.application_insights_connection_string != null
  }
```

But I'll update it for consistency